src/overlays/actors/ovl_Boss_Fd/z_boss_fd.c

```diff
@@ -21,51 +105,1974 @@ const ActorInit Boss_Fd_InitVars = {
+            Matrix_Scale(ptr->scale, ptr->scale, 1.0f, 1);
+
+            gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(gfxCtx, "../z_boss_fd.c", 0x1060),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(oGfxCtx->polyXlu.p++, D_0600D6E8);
+        }
+    }
+
+    CLOSE_DISPS(gfxCtx, "../z_boss_fd.c", 0x1066);
+}
+
+void BossFd_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BossFd* this = THIS;
+    s32 pad;
+
+    osSyncPrintf("FD DRAW START\n");
+    if (this->actionFunc != func_808CDE30) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd.c", 0x1079);
+        func_80093D18(globalCtx->state.gfxCtx);
+        if (this->damageFlashTimer & 2) {
+            oGfxCtx->polyOpa.p = Gfx_SetFog(oGfxCtx->polyOpa.p, 0xFF, 0xFF, 0xFF, 0, 0x384, 0x44B);
+        }
+
+        func_808D08F8(globalCtx, this);
+        oGfxCtx->polyOpa.p = func_800BC8A0(globalCtx, oGfxCtx->polyOpa.p);
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd.c", 0x1093);
+    }
+
+    osSyncPrintf("FD DRAW END\n");
+    func_808CF898(this->particles, globalCtx);
+    osSyncPrintf("FD DRAW END2\n");
+}
+
+s32 func_808D00A4(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    BossFd* this = THIS;
+
+    switch (limbIndex) {
+        case 1:
+            rot->y += 4000.0f + this->unk_C3C[0].x;
+            break;
+        case 2:
+            rot->y += this->unk_C3C[1].x;
+            rot->z += this->unk_C3C[1].y;
+            break;
+        case 3:
+            rot->y += this->unk_C3C[2].x;
+            rot->z += this->unk_C3C[2].y;
+            break;
+    }
+    if (this->skinSegments < limbIndex) {
+        *dList = NULL;
+    }
+    return 0;
+}
+
+s32 func_808D01C0(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    BossFd* this = THIS;
+
+    switch (limbIndex) {
+        case 1:
+            rot->y += -4000.0f + this->unk_C6C[0].x;
+            break;
+        case 2:
+            rot->y += this->unk_C6C[1].x;
+            rot->z += this->unk_C6C[1].y;
+            break;
+        case 3:
+            rot->y += this->unk_C6C[2].x;
+            rot->z += this->unk_C6C[2].y;
+            break;
+    }
+    if (this->skinSegments < limbIndex) {
+        *dList = NULL;
+    }
+    return 0;
+}
+
+s16 D_808D1AC0[] = { 0, 95, 90, 85, 80, 75, 70, 65, 60, 55, 50, 45, 40, 35, 30, 25, 20, 15, 10, 5 };
+s16 D_808D1AE8[] = { 0, 28, 26, 24, 22, 20, 18, 16, 14, 12, 10 }; // Unused? Seems to be for the mane.
+
+void func_808D02DC(GlobalContext* globalCtx, BossFd* this, Vec3f* manePos, Vec3f* maneRot, f32* maneScale, u8 mode) {
+    f32 sp140[] = { 0.0f, 10.0f, 17.0f, 20.0f, 19.5f, 18.0f, 17.0f, 15.0f, 15.0f, 15.0f };
+    f32 sp118[] = { 0.0f, 10.0f, 17.0f, 20.0f, 21.0f, 21.0f, 21.0f, 21.0f, 21.0f, 21.0f };
+    f32 spF0[] = { 0.4636457f, 0.33661291f, 0.14879614f, 0.04995025f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f };
```

Comment:
Those look like arctangent values of nicer numbers (eg `arctan(1/2) = 0.46364760900 (...)`) but I guess they were copypasted and I dont think the compiler can inline them anyway

Reply:
Tried it but the compiler didn't take it.